adagucserverEC/CServerParams.cpp

```diff
@@ -284,14 +283,10 @@ int CServerParams::makeLayerGroupName(CT::string *groupName,CServerConfig::XMLE_
   return 0;
 }
 
-bool CServerParams::isAutoResourceCacheEnabled(){
-  if(autoResourceCacheEnabled==-1){
-     autoResourceCacheEnabled = 0;
-    if(cfg->AutoResource.size()>0){
-      if(cfg->AutoResource[0]->attr.enablecache.equals("true"))autoResourceCacheEnabled = 1;
-    }
-  }
-  if(autoResourceCacheEnabled==1)return true;
+bool CServerParams::isAutoResourceCacheEnabled() const {
+
+  if (cfg->AutoResource.size() > 0)
+    return cfg->AutoResource[0]->attr.enablecache.equals("true");
```

Comment:
Previously at first a string compare and consecutive calls result only in an int compare. Not it is always a string compare, why?

Reply:
This was done so that this method could be made const. In the previous version private class member autoResourceCacheEnabled can be changed preventing this method from being changed to const.
I'm kind of hard pressed to believe this will have measurable impact, but we can change the line to

```
return cfg->AutoResource[0]->attr.enablecache.equals("true", 4);
```
Which will compare the length of the strings first. Since "false" is 5 bytes long, this will reduce the number of instructions from the original version if the value is usually "false". If the value is usually true, we can reverse the call:
```
return ! cfg->AutoResource[0]->attr.enablecache.equals("false", 5);
```

Personally, I think this kind of optimization is not worth the bother of reducing readability and robustness.